### ISSUE TITLE
Cache keys are evicted before the TTL

### ISSUE THREAD

**Issue by prasanjitm** [allegro/bigcache#267]
bigcache.NewBigCache(bigcache.Config{
		Shards:             128,
		LifeWindow:         30 * time.Minute,
		CleanWindow:        20 * time.Minute,
		MaxEntriesInWindow: 200 * 30 * 60,
		MaxEntrySize:       100,
		Verbose:            true,
		Hasher:             nil,
		HardMaxCacheSize:   500,
		OnRemove:           nil,
		OnRemoveWithReason: func(key string, entry []byte, reason bigcache.RemoveReason){
			logger.Info("Cache eviction", zap.String("key", key), zap.Uint32("reason", uint32(reason)))
		},
		Logger:             nil,
	})

This is my configuration for Bigcache. 

Problem statement:
The life window is set to 30 minutes. But, the cache keys are evicted before TTL(in 2/3 minutes). I have logged the OnRemoveWithReason. The reason is always 0 (Expired).

What am I missing?

**Comment by prasanjitm** [comment]
@janisz can you please help here

**Comment by cristaloleg** [comment]
Can you show what and how you're inserting key-values into the cache?

**Comment by prasanjitm** [comment]
I am storing data using Set(key string, entry []byte) function provided by bigcache
Two types of data are stored,

Type - 1 => 
length of the cache key = 39 characters, 
length of entry 87 bytes, 
rpm for this type of cache save is 30000/minute
Here cache key is (cachePrefix-{{variable}}), e.g abc-1, abc-2, abc-3 where "abc-" is the cachePrefix

Type - 2 => 
length of the cache key = 15 characters, 
length of entry 5154 bytes
e.g cache key is "def"

**Comment by prasanjitm** [comment]
@cristaloleg can you please help.

**Comment by cristaloleg** [comment]
Isn't problem is related that you're replacing your keys?

**Comment by prasanjitm** [comment]
Yes, maybe the keys are replaced by bigcache, but the reason should be in that case 1/2, not 0, as the TT has not reached. Where as the reason which is logged is 0.

I am using,
Bigcache version - v1.2.1
Golang version - golang:1.13

**Comment by janisz** [comment]
Could you prepare [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)

**Comment by prasanjitm** [comment]
observed that, when concurrent requests are made to update the a single cache key using Set(key string, entry []byte), where  multiple times the key is being set in memory, in that case the key is evicted multiple times after the TTL.
Actually key should be evicted once. 
Is this a known issue? r u working on this?

**Comment by janisz** [comment]
It's not a known issue. Could you prepare a code that reproduces this bug?

> when concurrent requests are made to update the a _single_ cache key using `Set(key string, entry []byte)`

Do you mean same key here?

> in that case the key is evicted multiple times _after_ the TTL

Do you mean before here?

**Comment by l-i-h-m** [comment]
your clean window is 20min
